proxy/src/main/java/com/wavefront/agent/channel/ChannelUtils.java

```diff
@@ -193,4 +198,27 @@ public static InetAddress getRemoteAddress(@Nonnull ChannelHandlerContext ctx) {
     InetSocketAddress remoteAddress = (InetSocketAddress) ctx.channel().remoteAddress();
     return remoteAddress == null ? null : remoteAddress.getAddress();
   }
+
+  /**
+   * Get path from the URI (i.e. discard everything after "?").
+   *
+   * @param uri URI to get path from
+   * @return path
+   */
+  public static String getPath(String uri) {
+    int paramsDelimiter = uri.indexOf('?');
```

Comment:
do you care if this is a URL? you can always do URI.create(uri).getPath();

Reply:
Agreed, simplified the whole thing